app/src/test/java/com/codesoom/assignment/application/ProductServiceTest.java

```diff
@@ -0,0 +1,143 @@
+package com.codesoom.assignment.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.codesoom.assignment.entity.Product;
+import com.codesoom.assignment.exception.ProductNotFoundException;
+import com.codesoom.assignment.repository.ProductRepository;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+@DataJpaTest
+@DisplayName("ProductService")
+public class ProductServiceTest {
+
+  @Autowired
+  private ProductRepository productRepository;
+  private ProductService productService;
+
+  private Product source;
+  private Product saved;
+
+  @BeforeEach
+  void setUp() {
+    productService = new ProductService(productRepository);
+  }
+
+  @Nested
+  @DisplayName("getList 메소드는")
+  class Describe_get_list {
+
+    @Nested
+    @DisplayName("저장된 product가 있다면")
+    class Context_existent_product {
+
+      @BeforeEach
+      void prepare() {
+        source = new Product(1L, "테스트", 2000, "");
+        productService.create(saved);
+      }
+
+      @Test
+      @DisplayName("ProductList를 리턴한다")
+      void it_returns_productList() {
+        assertThat(productService.getList()).hasSize(1);
+      }
+    }
+
+    @Nested
+    @DisplayName("저장된 product가 없다면")
+    class Context_non_existent_product {
+
+      void removeAll() {
+        productRepository.deleteAll();
+      }
+
+      @Test
+      @DisplayName("빈 productList를 리턴한다")
+      void it_returns_emptyList() {
+        removeAll();
+        assertThat(productService.getList()).hasSize(0);
+
+      }
```

Comment:
이렇게 `@BeforeEach`를 써서 "준비 코드"와 "검증 코드"를 분리해 보세요. 지금은 준비 코드가 한 줄짜리지만 나중에 복잡한 준비 코드를 만나게 되면 이런 식으로 분리하는 연습을 해보는 게 큰 도움이 될 거에요.

```suggestion
      @BeforeEach
      void removeAll() {
        productRepository.deleteAll();
      }

      @Test
      @DisplayName("빈 productList를 리턴한다")
      void it_returns_emptyList() {
        assertThat(productService.getList()).hasSize(0);

      }
```